datalad/distribution/get.py

```diff
@@ -385,20 +407,55 @@ def _recursive_install_subds_underneath(ds, recursion_limit, reckless, start=Non
         if sub.get('state', None) != 'absent':
             # dataset was already found to exist
             yield get_status_dict(
-                'install', ds=subds, status='notneeded', logger=lgr,
-                refds=refds_path)
-            # do not continue, even if an intermediate dataset exists it
+                'install', type='dataset', path=sub['path'], status='notneeded',
+                logger=lgr, refds=refds_path)
+            # still recurse, even if an intermediate dataset exists it
             # does not imply that everything below it does too
-        else:
-            # try to get this dataset
-            for res in _install_subds_from_flexible_source(
-                    ds,
-                    sub,
-                    reckless=reckless,
-                    description=description):
-                # yield everything to let the caller decide how to deal with
-                # errors
-                yield res
+            to_recurse_only.append(sub)
+            continue
+        # prepare to get this dataset
+        # compose a list of candidate clone URLs
+        clone_urls = _get_flexible_source_candidates_for_submodule(ds, sub)
+        if not clone_urls:
+            # yield error
+            yield get_status_dict(
+                action='install',
+                type='dataset',
+                ds=ds,
+                status='error',
+                message=(
+                    "Have got no candidates to install subdataset %s from.",
+                    op.relpath(sub['path'], start=sub['parentds'])),
+                logger=lgr,
+            )
+            continue
+        to_install.append((
+            sub,
+            clone_urls,
+            [src for name, src in clone_urls],
+        ))
+
+    from joblib import Parallel, delayed
+    with Parallel(n_jobs=1, verbose=10) as parallel:
+        # actually install them (in parallel)
+        clone_res = parallel(
+            delayed(unwind)(
+                clone_dataset,
```

Comment:
my wild guess is that it is the fact that you passing a function as an argument which causes grief to serialization, but it is just a wild guess

Reply:
nevermind, that is not it, tried
```diff
diff --git a/datalad/distribution/get.py b/datalad/distribution/get.py
index d0e2d4062..bbabd95e3 100644
--- a/datalad/distribution/get.py
+++ b/datalad/distribution/get.py
@@ -436,17 +436,16 @@ def _recursive_install_subds_underneath(ds, recursion_limit, reckless, start=Non
         ))
 
     from joblib import Parallel, delayed
-    with Parallel(n_jobs=1, verbose=10) as parallel:
+    with Parallel(n_jobs=5, verbose=10) as parallel:
         # actually install them (in parallel)
-        clone_res = parallel(
-            delayed(unwind)(
-                clone_dataset,
+        clone_res = list(parallel(
+            delayed(clone_dataset)(
                 urls,
                 sub['path'],
                 reckless=reckless,
                 description=description)
             for sub, url_info, urls in to_install
-        )
+        ))
     # fixup and register all subdatasets in the parent, must be done
     # in serial fashion to avoid locking issues
     for ti, res in zip(to_install, clone_res):
```
I think the easiest to start with would be a simple multithreaded parallelization which would not require any serialization of params etc